### PR TITLE
Report a user friendly message when the documentation is missing

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
None
### Summary of the issue:
When NVDA is run from source, if the documentation is missing, e.g. because it has not been built, there is a  user friendly message indicating that the documentation is missing when using context help. Though, when trying to open the documentation from NVDA's menu, there is no user friendly message; there is only an error in the log.

### Description of user facing changes

If the documentation is missing, a message pops up to report it to the user. In this case, we do not use `ui.message` as for context help, because the message is cut by the new focus being reported.

### Description of development approach

Factored the function to report that the doc is missing.

### Testing strategy:

Manual test, with and without built doc:
* Try to use context help
* NVDA menu -> Help: try to open User Guide, Key commands and Changes documentations.

### Known issues with pull request:

None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Added functionality to report missing documentation with a new user-friendly mechanism.
	- Introduced a new XML file for color mapping to enhance visual coherence in drawing applications.
	- Created a manifest XML file for the user guide, improving resource organization and accessibility.

- **Bug Fixes**
	- Streamlined error handling for missing user guides by utilizing a dedicated reporting function.

### CODE REVIEW

File: source/gui/__init__.py [hunk 1]
```diff
@@ -792,6 +792,13 @@ def _appendHelpSubMenu(self, frame: MainFrame) -> None:
 		# Translators: The label for the Help submenu in NVDA menu.
 		self.menu.AppendSubMenu(self.helpMenu, _("&Help"))
 
+	def _openDocumentationFile(self, fileName):
+		helpFile = getDocFilePath(fileName)
+		if helpFile is None:
+			reportNoDocumentation(fileName, useMsgBox=True)
+			return
+		os.startfile(helpFile)
```
[the remark this answers is not in the document]
Comment (CyrilleB79) [reply source/gui/__init__.py:800]: `getDocFilePath` has already checked that the path exists.

Are you thinking to any other possible specific exception that could be raised by `os.startfile`.

Note that in NVDA's Python code, the exceptions that may occur are already catched globally and logged in the log. So there is no point in writing a try clause catching all possible exceptions to log them if they occur here.

File: source/documentationUtils.py [hunk 2]
```diff
@@ -51,3 +54,23 @@ def getDocFilePath(fileName: str, localized: bool = True) -> Optional[str]:
...
+
+def reportNoDocumentation(fileName: str, useMsgBox: bool = False) -> None:
+	# Translators: Message reported upon action (e.g. context sensitive help, open documentation from NVDA menu)
+	noDocMessage = _("No documentation found.")
+	log.debugWarning(
+		f"Documentation not found ({fileName}): possible cause - running from source without building user docs.",
+	)
+	if useMsgBox:
+		from gui.message import messageBox
+		import wx
```
[the remark this answers is not in the document]
Comment (CyrilleB79) [reply source/documentationUtils.py]: Done in 02f42fab2462d715925abfd8021475485a0f03aa

File: source/gui/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CyrilleB79) [reply source/gui/__init__.py]: Done in 02f42fab2462d715925abfd8021475485a0f03aa

File: user_docs/fr/userGuide_fichiers/colorschememapping.xml
Comment (SaschaCowley) [inline user_docs/fr/userGuide_fichiers/colorschememapping.xml:1]: Please remove this file

File: user_docs/fr/userGuide_fichiers/filelist.xml
Comment (SaschaCowley) [inline user_docs/fr/userGuide_fichiers/filelist.xml:1]: Please remove this file

File: user_docs/fr/userGuide_fichiers/themedata.thmx
Comment (SaschaCowley) [inline user_docs/fr/userGuide_fichiers/themedata.thmx:1]: Please remove this file

File: source/gui/__init__.py [hunk 3]
```diff
@@ -737,13 +737,13 @@ def _appendHelpSubMenu(self, frame: MainFrame) -> None:
 		if not globalVars.appArgs.secure:
 			# Translators: The label of a menu item to open NVDA user guide.
 			item = self.helpMenu.Append(wx.ID_ANY, _("&User Guide"))
-			self.Bind(wx.EVT_MENU, lambda evt: os.startfile(getDocFilePath("userGuide.html")), item)
+			self.Bind(wx.EVT_MENU, lambda evt: self._openDocumentationFile("userGuide.html"), item)
```
Comment (SaschaCowley) [inline source/gui/__init__.py:740]: Can `evt` be changed to `_` to show it is unused?
```suggestion
			self.Bind(wx.EVT_MENU, lambda _: self._openDocumentationFile("userGuide.html"), item)
```

File: source/gui/__init__.py [hunk 4]
```diff
@@ -737,13 +737,13 @@ def _appendHelpSubMenu(self, frame: MainFrame) -> None:
...
 			# Translators: The label of a menu item to open the Commands Quick Reference document.
 			item = self.helpMenu.Append(wx.ID_ANY, _("Commands &Quick Reference"))
-			self.Bind(wx.EVT_MENU, lambda evt: os.startfile(getDocFilePath("keyCommands.html")), item)
+			self.Bind(wx.EVT_MENU, lambda evt: self._openDocumentationFile("keyCommands.html"), item)
```
Comment (SaschaCowley) [inline source/gui/__init__.py:743]: See comment at line 740.

File: source/gui/__init__.py [hunk 5]
```diff
@@ -737,13 +737,13 @@ def _appendHelpSubMenu(self, frame: MainFrame) -> None:
...
 			# Translators: The label for the menu item to open What's New document.
 			item = self.helpMenu.Append(wx.ID_ANY, _("What's &new"))
-			self.Bind(wx.EVT_MENU, lambda evt: os.startfile(getDocFilePath("changes.html")), item)
+			self.Bind(wx.EVT_MENU, lambda evt: self._openDocumentationFile("changes.html"), item)
```
Comment (SaschaCowley) [inline source/gui/__init__.py:746]: See comment at line 740.

File: source/gui/contextHelp.py [hunk 6]
```diff
@@ -40,13 +40,7 @@ def showHelp(helpId: str):
...
 	helpFile = documentationUtils.getDocFilePath("userGuide.html")
 	if helpFile is None:
-		# Translators: Message shown when trying to display context sensitive help,
-		# indicating that	the user guide could not be found.
-		noHelpMessage = _("No user guide found.")
-		log.debugWarning(
-			"No user guide found: possible cause - running from source without building user docs",
-		)
-		queueHandler.queueFunction(queueHandler.eventQueue, ui.message, noHelpMessage)
+		documentationUtils.reportNoDocumentation("userGuide.html")
```
Comment (SaschaCowley) [inline source/gui/contextHelp.py:43]: Should this use a message box as well, or would this conflict with context help being requested from inside modal dialogs in NVDA?

**Review by SaschaCowley — CHANGES_REQUESTED** [review]
Thanks for this work Cyrille. It looks like you've accidentally included a few files that shouldn't be in the repo. I've also suggested a few small changes, but this looks good overall to me.

File: source/gui/__init__.py (hunk 3 above)
Comment (CyrilleB79) [reply source/gui/__init__.py:740]: I'd rather avoid to use "_" (underscore character) for unused variables in our code.

Indeed it is already used as a gettext translation function. IMO, for clarity, it's better not to use the same variable name for two distinctive usages, even if the scope is not the same.

And what if you want to use translated content in a lambda function taking unused arguments in the future?

Comment (SaschaCowley) [reply source/gui/__init__.py:740]: > And what if you want to use translated content in a lambda function taking unused arguments in the future?

You could still use `gettext` directly.

Nevertheless, that's a very good point. I'm personally not a huge fan of having named unused variables, but it seems there's no very good alternative here.

File: source/gui/contextHelp.py (hunk 6 above)
Comment (CyrilleB79) [reply source/gui/contextHelp.py:43]: Currently, we are using `ui.message` for context help.

Taking into account that the current solution is working and that the use case of missing documentation is quite rare (so rarely tested), I have not investigated the message box solution and have opted to keep the same design as today.

Using a message box for context help seems to be working. I am not totally sure if specific precautions should be taken and which specific cases should be tested regarding modal dialogs and restart race conditions. Maybe @seanbudd can help for this?

File: source/gui/__init__.py (hunk 3 above)
Comment (CyrilleB79) [reply source/gui/__init__.py:740]: Moreover here, naming it `evt` also conveys the fact that this function is passed the event (even if this event is not used).

If you are concerned by the fact that we do not indicate that the variable is unused, I may name it `unusedEvt` instead, but that's a bit ugly and probably not necessary for a so short function body.

Comment (SaschaCowley) [reply source/gui/__init__.py:740]: No, I think having it named `evt` is best here. Sorry for the fuss

File: source/gui/contextHelp.py (hunk 6 above)
Comment (SaschaCowley) [reply source/gui/contextHelp.py:43]: I think it should be fine to stick with `ui.message` in these cases then. If users have problems with the current approach, a new issue can be opened for it and we can revisit using a message box instead.

**Review by SaschaCowley — APPROVED** [review]

**Pull request merged**